internal/virtualcircuit/create.go

```diff
@@ -0,0 +1,157 @@
+package virtualcircuit
+
+import (
+	"context"
+	"fmt"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) CreateVirtualCricuit() *cobra.Command {
+	var (
+		connectionID string
+		portID       string
+		name         string
+		description  string
+		projectID    string
+		vnid         string
+		vrf          string
+		subnet       string
+		customerIP   string
+		metalIP      string
+		md5          string
+		peerAsn      int
+		speed        int
+		nnVlan       int
+		tags         []string
+	)
+
+	createVirtualCricuitCmd := &cobra.Command{
+		Use:   `create  [-i connection_id] [-p port_id] [-P <project_id> ] --n <name> [-d <description>] [-v <vlanID> ] [-t <tags> ] [-vlan <vlan> ] [-speed <speed> ]`,
+		Short: "Creates an create-virtual-circuit for specific interconnection.",
+		Long:  "Creates an create-virtual-circuit for specific interconnection",
+		Example: `  # Creates a new virtual-circuit named "interconnection": 
+  metal virtual-circuit create [-i connection_id] [-p port_id] [-P <project_id> ] --n <name> [-d <description>] [-v <vlanID> ] [-t <tags> ] [-vlan <vlan> ] [-speed <speed> ]
+  
+  metal virtual-circuit create [-i connection_id] [-p port_id] [-P <project_id> ] -n <name> [-m <metro>] [-r <redundancy>] [-t "shared" ] -T <service_token_type> -v <vrfs>
+  metal virtual-circuit create [-i connection_id] [-p port_id] [-P <project_id> ] -n <name> [-m <metro>] [-r <redundancy>] [-t "shared" ] -T <service_token_type> -v <vrfs>`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			var virtualCircuitCreateInput metal.VirtualCircuitCreateInput
+
+			if vrf != "" {
+				vrfVirtualCircuitCreateInput := metal.NewVrfVirtualCircuitCreateInputWithDefaults()
+
+				if projectID != "" {
+					vrfVirtualCircuitCreateInput.SetProjectId(projectID)
```

Comment:
Unless `SetProjectId` is converting a `string` to a `*string` in the request object, we don't need the condition here. It does nothing.  The same applies for all of these conditions. Especially for the `Create` action.

Reply:
I see that there is a mix of string->string, and string->*string / int->*int.
https://github.com/equinix-labs/metal-go/blob/main/metal/v1/model_vlan_virtual_circuit_create_input.go

For the Setters where the type is not converted, the `if` statement is not needed.

Reply:
ProjectID for example, does not need the conditional.